core/src/main/java/bisq/core/offer/takeoffer/TakeOfferModel.java

```diff
@@ -0,0 +1,293 @@
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.bitcoinj.core.Coin.ZERO;
+import static org.bitcoinj.core.Coin.valueOf;
+
+@Slf4j
+public class TakeOfferModel implements Model {
+    // Immutable
+    private final AccountAgeWitnessService accountAgeWitnessService;
+    private final BtcWalletService btcWalletService;
+    private final FeeService feeService;
+    private final OfferUtil offerUtil;
+    private final PriceFeedService priceFeedService;
+
+    // Mutable
+    @Getter
+    private AddressEntry addressEntry;
+    @Getter
+    private Coin amount;
+    @Getter
+    private boolean isCurrencyForTakerFeeBtc;
+    private Offer offer;
+    private PaymentAccount paymentAccount;
+    @Getter
+    private Coin securityDeposit;
+    private boolean useSavingsWallet;
+
+    // 260 kb is typical trade fee tx size with 1 input, but trade tx (deposit + payout)
+    // are larger so we adjust to 320.
+    private final int feeTxSize = 320;
+    private Coin txFeePerByteFromFeeService;
+    @Getter
+    private Coin txFeeFromFeeService;
+    @Getter
+    private Coin takerFee;
+    @Getter
+    private Coin totalToPayAsCoin;
+    @Getter
+    private Coin missingCoin = ZERO;
+    @Getter
+    private Coin totalAvailableBalance;
+    @Getter
+    private Coin balance;
+    @Getter
+    private boolean isBtcWalletFunded;
+    @Getter
+    private Volume volume;
+
+    @Inject
+    public TakeOfferModel(AccountAgeWitnessService accountAgeWitnessService,
+                          BtcWalletService btcWalletService,
+                          FeeService feeService,
+                          OfferUtil offerUtil,
+                          PriceFeedService priceFeedService) {
+        this.accountAgeWitnessService = accountAgeWitnessService;
+        this.btcWalletService = btcWalletService;
+        this.feeService = feeService;
+        this.offerUtil = offerUtil;
+        this.priceFeedService = priceFeedService;
+    }
+
+    public void initModel(Offer offer,
+                          PaymentAccount paymentAccount,
+                          boolean useSavingsWallet) {
+        this.clearModel();
+        this.offer = offer;
+        this.paymentAccount = paymentAccount;
+        this.addressEntry = btcWalletService.getOrCreateAddressEntry(offer.getId(), OFFER_FUNDING);
+        validateModelInputs();
+
+        this.useSavingsWallet = useSavingsWallet;
+        this.amount = valueOf(Math.min(offer.getAmount().value, getMaxTradeLimit()));
+        this.securityDeposit = offer.getDirection() == SELL
+                ? offer.getBuyerSecurityDeposit()
+                : offer.getSellerSecurityDeposit();
+        this.isCurrencyForTakerFeeBtc = offerUtil.isCurrencyForTakerFeeBtc(amount);
+        this.takerFee = offerUtil.getTakerFee(isCurrencyForTakerFeeBtc, amount);
+
+        calculateTxFees();
+        calculateVolume();
+        calculateTotalToPay();
```

Comment:
This is already done in calculateTxFees

Reply:
Resolved by commit 1f3554e in PR https://github.com/bisq-network/bisq/pull/4679.